src/setup-gcloud.ts

```diff
@@ -100,17 +100,33 @@ export async function run(): Promise<void> {
         credsPath = path.join(githubWorkspace, uniqueName);
       }
 
-      const serviceAccountKeyObj = parseServiceAccountKey(serviceAccountKey);
-      await writeSecureFile(
-        credsPath,
-        JSON.stringify(serviceAccountKeyObj, null, 2), // Print to file as string w/ indents
-      );
+      // If explicit SA input, parse it, write to disk and set GCLOUD_PROJECT.
+      if (serviceAccountKey) {
+        const serviceAccountKeyObj = parseServiceAccountKey(serviceAccountKey);
+        await writeSecureFile(
+          credsPath,
+          JSON.stringify(serviceAccountKeyObj, null, 2), // Print to file as string w/ indents
+        );
+        core.exportVariable('GCLOUD_PROJECT', serviceAccountKeyObj.project_id);
```

Comment:
I think we should only set the project ID here if the `project_id` input wasn't given above. If the user explicitly set a project_id, we should prefer that over the SA's project. Probably be good to compare the project IDs and throw a warning if they are different too

Reply:
Done and added a warning. In the earlier flow we were setting `GCLOUD_PROJECT` to `serviceAccountKeyObj.project_id` and then re exporting `GCLOUD_PROJECT` as `projectID` if set. Hopefully the newer flow is clearer.